content/ko/docs/tasks/tls/certificate-rotation.md

```diff
@@ -0,0 +1,79 @@
+---
+
+
+
+title: Kubelet의 인증서 갱신 구성
+content_type: task
+---
+
+<!-- overview -->
+이 페이지는 kubelet에 대한 인증서 갱신을 활성화하고 구성하는 방법을 보여준다.
+
+
+{{< feature-state for_k8s_version="v1.8" state="beta" >}}
+
+## {{% heading "prerequisites" %}}
+
+
+* 쿠버네티스 1.8.0 버전 혹은 그 이상의 버전이 요구됨
+
+
+
+<!-- steps -->
+
+## 개요
+
+kubelet은 Kubernetes API 인증을 위해 인증서를 사용한다.
+기본적으로 이러한 인증서는 1년 만기로 발급되므로
+너무 자주 갱신할 필요는 없다.
+
+Kubernetes 1.8에는 [kubelet 인증서
+갱신] (/docs/reference/command-line-tools-reference/kubelet-tls-bootstrapping/) 를 포함하며,
+현재 인증서가 만료됨에 따라
+새로운 키를 자동으로 생성하고 Kubernetes API에서 새로운 인증서를 요청하는 베타 기능이다.
+새로운 인증서를 사용할 수 있게 되면
+Kubernetes API에 대한 연결을 인증하는데 사용된다.
+
+## 클라이언트 인증서 갱신 가능
+
+`kubelet` 프로세스는 현재 사용중인 인증서가 만료됨에 따라
+kubelet이 자동으로 새 인증서를 요청할지 여부를 제어하는
+`--rotate-certificates` 인자를 허용한다.
+인증서 갱신은 베타 기능이므로 기능 플래그는
+`--feature-gates = RotateKubeletClientCertificate=true` 를 사용하여 활성화해야 한다.
+
+
+`kube-controller-manager` 프로세스는 얼마나 오랜기간 인증서가 유효한지를 제어하는
+`--experimental-cluster-signing-duration` 인자를
+허용한다.
+
+## 인증서 갱신 구성에 대한 이해
+
+kubelet이 시작할 때 부트 스트랩 (`--bootstrap-kubeconfig` 플래그를 사용)
+을 구성하면 초기 인증서를 사용하여 Kubernetes API에 연결하고
+인증서 서명 요청을 발행한다.
+다음을 사용하여 인증서 서명 요청 상태를 볼 수 있다:
+
+```sh
+kubectl get csr
+```
+
+초기에 노드의 kubelet에서 인증서 서명 요청은 `Pending` 상태이다.
+인증서 서명 요청이 특정 기준을 충족하면 컨트롤러 관리자가
+자동으로 승인한 후 상태가 `Approved` 가 된다.
+다음으로, 컨트롤러 관리자는
+`--experimental-cluster-signing-duration` 파라미터에 의해 지정된 기간 동안
+발행된 인증서에 서명하고
+서명된 인증서는 인증서 서명 요청에 첨부된다.
+
+kubelet은 Kubernetes API로 서명된 인증서를 가져와서
+`--cert-dir`에 지정된 위치에 디스크에 기록한다.
+그런다음 kubelet은 Kubernetes API에 연결해서 새로운 인증서를 사용한다.
+
+서명된 인증서가 만료되면 kubelet은 Kubernetes API를 사용하여
+새로운 인증서 서명 요청을 자동으로 발행한다.
+또한, 컨트롤러 관리자는 인증서 요청을 자동으로 승인하고
+서명된 인증서를 인증서 서명 요청에 첨부한다.
+kubelet은 Kubernetes API로 서명된 새로운 인증서를 가져와서 디스크에 쓴다.
+그런다음 Kubernetes API를 가지고 있는 연결을 업데이트하여
+사용 중인 새로운 인증서를 재 연결한다.
```

Comment:
```suggestion
kubelet은 쿠버네티스 API로 서명된 새로운 인증서를 가져와서 디스크에 쓴다.
그런 다음 새로운 인증서를 사용한 재연결을 위해서
가지고 있는 쿠버네티스 API로의 연결을 업데이트 한다.





```
원문에 조금 더 가깝게 수정해 보았습니다. 참고하시기 바랍니다.
Then it will update the connections it has to the
Kubernetes API to reconnect using the new certificate.

Reply:
감사합니다~ 수정 완료했습니다